### ISSUE TITLE
Remove sampling from `forward`

### ISSUE THREAD

**Issue by pgniewko** [pgniewko/gt-pyg#104]
**NOTE**: This issue invalidates #102 
---------------------

`forward` samples `pred = mu + exp(0.5*log_var)*eps` (`model.py:320`) but nothing trains `log_var`. Under a point loss the optimum is `sigma=0`, so it slides to the `-10` clamp (`model.py:317`) where gradient is 0 and the head dies. 

## Fix

```python
mu = self.mu_mlp(g)
log_var = torch.clamp(self.log_var_mlp(g), min=-10.0, max=10.0)
pred = mu
```

Drop `zero_var` (no caller uses it). Update `test_model.py:227` and the docstrings.
Keep the clamp and the `(pred, log_var)` return.

## Docs

Notebooks aren't version-compatible, so NLL training stays caller-side. 

Add to `README.md`:

````markdown
### Uncertainty estimates

`log_var` gets no gradient from a plain regression loss:

```python
loss = your_loss(pred, y) + F.gaussian_nll_loss(pred.detach(), y, log_var.exp())
```

`.detach()` keeps the NLL off the mean: NLL trains only `log_var`, and `mu` stays on the existing point-loss mix.
 Masked: `reduction='none'`, then `(nll * valid).sum() / valid.sum().clamp_min(1)`. 
For data with missing values, as in notebooks, `valid = mask.bool() & torch.isfinite(y) & torch.isfinite(pred)`.
At eval `sigma = (0.5*log_var).exp()` is a real error bar.
```
````
